driver/adfs_proxy.h

```diff
@@ -30,40 +30,61 @@
 #ifndef __ADFS_PROXY__
 #define __ADFS_PROXY__
 
+#include <regex>
 #include <unordered_map>
 #include "auth_util.h"
+#include "saml_http_client.h"
+#include "saml_util.h"
+
+namespace ADFS_REGEX {
+const std::regex FORM_ACTION_PATTERN(R"#(<form.*?action=\"([^\"]+)\")#", std::regex_constants::icase);
+const std::regex SAML_RESPONSE_PATTERN("\"SAMLResponse\"\\W+value=\"(.*?)\"(\\s*/>)", std::regex_constants::icase);
+const std::regex URL_PATTERN(R"#(^(https)://[-a-zA-Z0-9+&@#/%?=~_!:,.']*[-a-zA-Z0-9+&@#/%=~_'])#",
+                             std::regex_constants::icase);
+const std::regex INPUT_TAG_PATTERN(R"#(<input id=(.*))#", std::regex_constants::icase);
+}  // namespace ADFS_REGEX
```

Comment:
Minor style suggestion: I think indenting this inside the `namespace` block may make reading this a little easier and eliminates the need for the end comment.